src/Language/Haskell/Liquid/Bare/OfType.hs

```diff
@@ -100,9 +102,10 @@ rtypePredBinds = map uPVar . ty_preds . toRTypeRep
 ofBRType :: (PPrint r, Reftable r)
          => (SourcePos -> RTAlias RTyVar SpecType -> [BRType r] -> r -> BareM (RRType r))
          -> (r -> BareM r)
+         -> (Expr -> BareM Expr)
          -> BRType r
          -> BareM (RRType r)
-ofBRType appRTAlias resolveReft
+ofBRType appRTAlias resolveReft resolveExpr
```

Comment:
Ah quite right, I saw the `Loc` in the `RApp` case and I guess I stopped reading :)

I suspect the `Loc` is used for error reporting, in which case we shouldn't just remove it outright. (There's another issue here somewhere that talks about adding `Loc`s to all of the datatypes we use and making `Located` a type-class, but I can't find it in github's mobile site..)